### PR TITLE
fix beneficiary issue

### PR DESCRIPTION
**Pull request by rlan35**

### CODE REVIEW

File: cmd/harmony/main.go [hunk 1]
```diff
@@ -295,7 +294,9 @@ func setupConsensusAndNode(nodeConfig *nodeconfig.ConfigType) *node.Node {
 		return currentConsensus.PubKey, nil
 	})
 
-	currentConsensus.SelfAddress = common.ParseAddr(initialAccount.Address)
+	if initialAccount.Address != "" { // staking validator doesn't have to specify ECDSA address
+		currentConsensus.SelfAddress = common.ParseAddr(initialAccount.Address)
```
Comment (chaosma) [inline cmd/harmony/main.go]: Is initialAccount.Address always bls address?

File: node/node_newblock.go [hunk 2]
```diff
@@ -83,14 +84,27 @@ func (node *Node) proposeNewBlock() (*types.Block, error) {
...
+		err         error
+	)
 
 	// After staking, all coinbase will be the address of bls pub key
 	if node.Blockchain().Config().IsStaking(node.Worker.GetCurrentHeader().Epoch()) {
 		addr := common.Address{}
-		blsPubKeyBytes := node.Consensus.PubKey.GetAddress()
-		addr.SetBytes(blsPubKeyBytes[:])
-		coinbase = addr
+		if bytes.Compare(coinbase[:], addr[:]) == 0 { // empty SelfAddress means it's a staking validator
```
Comment (chaosma) [inline node/node_newblock.go]: what if coinbase not equal bls address (unless selfaddress always equal bls address in this case)?

File: cmd/harmony/main.go (hunk 1 above)
Comment (rlan35) [reply cmd/harmony/main.go]: Nop. I changed it to always be the ECDSA address or empty "" if it's staking validator.

File: node/node_newblock.go (hunk 2 above)
Comment (rlan35) [reply node/node_newblock.go]: Here it's checking whether SelfAddress is empty. If it's empty, it means it's staking validator, and the ECDSA address needs to be looked up.

**Pull request merged**